### ISSUE TITLE
SEGFAULT when uv_udp_recv_stop or uv_read_stop in uv_alloc_cb

### ISSUE THREAD

**Issue by renatomaia** [libuv/libuv#2399]
* **Version**: 1.30.2
* **Platform**: `Linux host 4.4.0-157-generic #185-Ubuntu SMP Tue Jul 23 09:17:01 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux`

The code in the attached files ([segfault_udp.c](https://github.com/libuv/libuv/files/3444668/segfault_udp.c.txt), [segfault_tcp.c](https://github.com/libuv/libuv/files/3444670/segfault_tcp.c.txt)) causes a SEGFAULT inside libuv when it stops reading from a  `uv_udp_t` or `uv_tcp_t` by invoking `uv_udp_recv_stop` or `uv_read_stop` inside the `uv_alloc_cb` callback.

Also see: https://groups.google.com/d/topic/libuv/QWg9XLt-fuw/discussion

**Comment by santigimeno** [comment]
Yes, calling stop in the `alloc_cb` cannot work as the `read_cb` is set to `NULL` and later is being accessed. I would probably add an assertion in the code to check for the validity of the `read_cb` and a note in the documentation of the stop functions to warn users not to use them in the `alloc_cb`.

**Comment by saghul** [comment]
Or we could make it legal by not calling the read callback, right? I guess this is only applicable in case you try to allocate memory, fail and as a result don't want to read anymore...

**Comment by santigimeno** [comment]
> Or we could make it legal by not calling the read callback, right?

Yeah, that makes more sense

**Comment by juanarbol** [comment]
I'm investigating this one
